ICAFIX/scripts/RecleanClassify.sh

```diff
@@ -0,0 +1,284 @@
+    fi
+
+    if command -v singularity &> /dev/null; then
+        log_Msg "Singularity is installed."
+    else
+        log_Err_Abort "Singularity is not installed or not in PATH."
+    fi
+    UseLocalPython="FALSE"
+    singularity_command=(singularity exec --bind "$PythonSingularityMountPath" "$PythonSingularity" python3)
+else
+    # native env using conda
+    if [ ! -f "$PythonInterpreter" ]; then
+        PythonInterpreter="python3" # default native env
+    fi
+fi
+
+if [[ "$VoteThresh" == "" ]]; then
+    count=$(echo "$ModelToUse" | awk -F'@' '{print NF}')
+    VoteThresh=$((count - 1))
+fi
+
+IFS='@' read -a fMRINamesArray <<<"$fMRINames"
+
+tempfiles_create fMRIList_XXXXXX.txt fMRIListName
+
+# check SR or MR FIX
+if [ ! ${MRFixConcatName} = "" ] ; then
+    echo "${MRFixConcatName}" >> "$fMRIListName"
+    fMRINameToUse=${MRFixConcatName}
+else
+    fMRINameToUse=""
+    for fMRIName in "${fMRINamesArray[@]}"
+    do
+        echo "${fMRIName}" >> "$fMRIListName"
+        fMRINameToUse+=" $fMRIName"
+    done
+    # Remove the leading space
+    fMRINameToUse="${fMRINameToUse:1}"
+fi
+
+# check if FIX features are generated (csv with 181 features)
+for fMRIName in ${fMRINameToUse} ; do
+    FixFeaturePath="${StudyFolder}/${Subject}/MNINonLinear/Results/${fMRIName}/${fMRIName}_hp${HighPass}.ica/fix/features.csv"
+    if [[ ! -e "$FixFeaturePath" ]]; then
+        log_Err_Abort "$FixFeaturePath is not doesn't exist, make sure ICA+FIX is applied to this subject: ${Subject}, fMRI run: ${fMRIName}"
+    fi
+done
+
+# Preprocess step 1: Merge and dropout
+Caret7_Command="wb_command"
+
+#TODO: enable multiple resolutions like macaque (1.25mm resolution) or 7T (1.6mm resolution)
+FinalfMRIResolution="2"
+SmoothingFWHM="2"
+BrainOrdinatesResolution="2"
+Flag="FALSE" #FALSE=Don't fix zeros, TRUE=Fix zeros
+DeleteIntermediates="TRUE" #TRUE/FALSE
+
+CorticalLUT="$HCPPIPEDIR/global/config/FreeSurferCorticalLabelTableLut.txt"
+SubCorticalLUT="$HCPPIPEDIR/global/config/FreeSurferSubcorticalLabelTableLut.txt"
+
+if [ ! ${MRFixConcatName} = "" ] ; then
+    DropOutSubSTRING=""
+    for fMRIName in "${fMRINamesArray[@]}" ; do
+        if [ -e ${StudyFolder}/${Subject}/MNINonLinear/Results/${fMRIName}/${fMRIName}_dropouts.nii.gz ] ; then
+        DropOutSubSTRING=`echo "${DropOutSubSTRING}${StudyFolder}/${Subject}/MNINonLinear/Results/${fMRIName}/${fMRIName}_dropouts.nii.gz "`
+        fi
+    done
+    if [ ! -z "${DropOutSubSTRING}" ] ; then
+        fslmerge -t ${StudyFolder}/${Subject}/MNINonLinear/Results/${MRFixConcatName}/${MRFixConcatName}_dropouts.nii.gz ${DropOutSubSTRING}
+        fslmaths ${StudyFolder}/${Subject}/MNINonLinear/Results/${MRFixConcatName}/${MRFixConcatName}_dropouts.nii.gz -Tmean ${StudyFolder}/${Subject}/MNINonLinear/Results/${MRFixConcatName}/${MRFixConcatName}_dropouts.nii.gz
+        WorkingDirectory="/tmp/${Subject}"
```

Comment:
Please give this a better identifier, in case it gets left behind and someone has to figure out what it is.  Perhaps `/tmp/RecleanClassify_"$Subject"`?